### PR TITLE
Wait for republish in MQTT integration tests for AWS IoT Core

### PR DESCRIPTION
**Pull request by yourslab**
Description
-----------

AWS IoT Core takes some time before resending an unack'd PUBLISH, so a delay is added before resuming the persistent session.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/integration_test/core_mqtt_system_test.c [hunk 1]
```diff
@@ -1562,6 +1569,17 @@ void Restore_Session_Duplicate_Incoming_Publish_Qos1()
...
 
+    if( testingAgainstAWS )
+    {
+        /* AWS IoT Core takes at least 30 seconds before resending a PUBLISH. */
+        vTaskDelay( pdMS_TO_TICKS( AWS_IOT_CORE_REPUBLISH_INTERVAL_MS ) );
+
+        /* When the resent PUBLISH is received, the DUP flag will not be set to 1.
+         * Therefore, the packet ID of the original PUBLISH must be cleared from
+         * the incoming publish records array. */
+        context.incomingPublishRecords[ 0 ].packetId = MQTT_PACKET_ID_INVALID;
```
Comment (aggarw13) [inline tests/integration_test/core_mqtt_system_test.c:1580]: Why is this required for the `MQTTContext_t`. According to the spec, even if the DUP flag is set, it does not mean that the receiving end had received the original PUBLISH message..

Comment (yourslab) [reply tests/integration_test/core_mqtt_system_test.c:1580]: IoT Core will not set the DUP flag in the resent publish. However, during the first time we receive the publish, the state array is updated. Therefore, when we receive it again and the DUP flag is not set, the library will return a `MQTTStateCollision` error.

**Review by aggarw13 — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]
